include/openthread-types.h

```diff
@@ -207,6 +207,24 @@ typedef struct otLinkModeConfig
 } otLinkModeConfig;
 
 /**
+ * This enumeration represents flags that indicate what configuration or state has changed within OpenThread.
+ *
+ */
+enum
+{
+    OT_IP6_ADDRESS_ADDED    = 1 << 0,  ///< IPv6 address was added
+    OT_IP6_ADDRESS_REMOVED  = 1 << 1,  ///< IPv6 address was removed
```

Comment:
I'm fine with how this is set up at the moment (since I'm eager to play around with this code!), but we should think about a way to actually allow Spinel to emit the proper individual `PROP_VALUE_INSERTED`/`PROP_VALUE_REMOVED` events. I'll admit, given the current architecture, it isn't immediately obvious to me what is the best way to do that. 

One way would be to have "inserted" and "removed" bitmasks, with each bit representing one of the slots in the array of address structures. It would require a few API changes, but it might be worth considering for a separate pull request.